site/app/libraries/database/DatabaseQueries.php

```diff
@@ -452,6 +452,55 @@ public function getCategoriesIdForThread($thread_id) {
         return $categories_list;
     }
 
+    public function splitPost($post_id, $title, $categories_ids) {
+        $this->course_db->query("SELECT * from posts where id = ?", array($post_id));
+        $posts = $this->course_db->rows();
+        $old_thread_id = -1;
+        $thread_id = -1;
+        // Safety measure in case the database is bad for some reason
+        $counted_posts = array();
+        if (count($posts) > 0) {
+            $post = $posts[0];
+            if ($post["parent_id"] != -1) {
+                $old_thread_id = $post["thread_id"];
+                $this->course_db->query("SELECT id from threads where merged_post_id = ?", array($post_id));
+                $thread_id = $this->course_db->rows();
+                if (count($thread_id) > 0) {
+                    $thread_id = $thread_id[0]["id"];
+                    $this->course_db->query("UPDATE threads set merged_thread_id=-1, merged_post_id=-1 where id=?", array($thread_id));
+                    $this->course_db->query("DELETE FROM thread_categories where thread_id=?", array($thread_id));
+                }
+                else {
+                    $this->course_db->query("INSERT INTO threads (title, created_by, is_visible, lock_thread_date) VALUES (?, ?, ?, ?)", array($title, $post["author_user_id"], true, null));
+                    $this->course_db->query("SELECT MAX(id) as max_id from threads where title=? and created_by=?", array($title, $post["author_user_id"]));
+                    $thread_id = $this->course_db->rows()[0]["max_id"];
+                }
+                foreach ($categories_ids as $id) {
+                    $this->course_db->query("INSERT INTO thread_categories (thread_id, category_id) VALUES (?, ?)", array($thread_id, $id));
+                }
+                while (count($posts) > 0) {
+                    $check_posts = array();
+                    $str = "";
+                    for ($i = 0; $i < count($posts); $i++) {
```

Comment:
nit: It does not look like `$i` is being used anywhere other than accessing `$posts`. Let's use a foreach here.